### PR TITLE
First-pass Jenkins template

### PR DESCRIPTION
**Pull request by ferricoxide**
runs to green

### CODE REVIEW

File: Templates/make_jenkins_master_standalone.tmplt.json [hunk 1]
```diff
@@ -618,115 +620,129 @@
...
+              "    gpgkey: ",
+                { "Ref" : "JenkinsRepoKeyURL" },
+              "\n",
+              "\n",
+              "runcmd:\n",
+              "  - |\n",
+              "      \n",
+              "      # Helper function\n",
+              "      function error_exit {\n",
+              "         /opt/aws/bin/cfn-signal -e 0 --stack ",
```
Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: This needs to be `-e 1` to fail the resource on error. `-e 0` will mark the resource as a success.

File: Templates/make_jenkins_master_standalone.tmplt.json [hunk 2]
```diff
@@ -618,115 +620,129 @@
...
+              "      }\n",
+              "      \n",
+              "      # Temp-disable SELinux\n",
+              "      setenforce 0 || error_exit 'Could not change SEL-mode'\n",
+              "      \n",
+              "      # Try to get firewalld ready for commands\n",
+              "      firewall-offline-cmd --enabled\n",
+              "      \n",
+              "      # Add firewall exceptions\n",
+              "      firewall-cmd --zone=public --add-port=80/tcp --permanent\n",
```
Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: I vaguely remember differences between AMI starting points where sometimes firewalld wasn't pre-installed. Might not be an issue with the intended starting points for this template, but might be a nice from a belt and suspenders perspective to install firewalld prior to trying to use it?

File: Templates/make_jenkins_master_standalone.tmplt.json [hunk 3]
```diff
@@ -60,6 +60,11 @@
       "Type": "String",
       "AllowedPattern": "^[a-z][a-z0-9-]*$"
     },
+    "BackupCronURL": {
```
Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: Add the new param to the parameter map?

File: Templates/make_jenkins_master_standalone.tmplt.json (hunk 2 above)
Comment (ferricoxide) [inline Templates/make_jenkins_master_standalone.tmplt.json]: You're thinking of `iptables`. The `firewalld` is part of Core.
~~~
$ yum groupinfo -v core | grep -i firewall
    firewalld-0.4.3.2-8.1.el7_3.2.noarch                             @updates
~~~

File: Templates/make_jenkins_master_standalone.tmplt.json (hunk 3 above)
Comment (ferricoxide) [inline Templates/make_jenkins_master_standalone.tmplt.json]: Updated and re-pushed...

File: Templates/make_jenkins_master_standalone.tmplt.json [hunk 4]
```diff
@@ -393,119 +399,115 @@
       "Type": "AWS::EC2::Instance",
```
Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: The resource name is "CfnLaunchConfig" but the type is an EC2 instance. That's a little confusing. Is this supposed to be used with an autoscale group? Or is that a future change?

File: Templates/make_jenkins_master_standalone.tmplt.json [hunk 5]
```diff
@@ -618,115 +620,131 @@
...
+                        { "Ref" : "BackupFolder" },
+                        "/backups/sync/JENKINS_HOME/ /var/lib/jenkins\" || ",
+                        "echo \"Fresh Install: No restorable S3 content found.\"\n",
+              "      \n",
+              "      # Start Jenkins\n",
+              "      systemctl enable jenkins\n",
+              "      systemctl start jenkins\n",
+              "      \n",
+              "      # Install Jenkins backup cron-job\n",
               "      test -f /tmp/jenkins_cron && install -b -m 0600 -o ",
```
Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: Won't `test -f` exit 0 if the file exists? Shouldn't the `&&` be `||` to create the cron file if the file does not exist?

File: Templates/make_jenkins_master_standalone.tmplt.json (hunk 2 above)
Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: I'm pretty sure I ran into when working EL7 support for spel, probably the RHEL AMIs. I'm not super worried about it here, though.

Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: Yeah, the RHEL AMIs were the problem:

```
# which firewall-cmd
/usr/bin/which: no firewall-cmd in (/usr/local/sbin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin)
# which firewall-offline-cmd
/usr/bin/which: no firewall-offline-cmd in (/usr/local/sbin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin)
# rpm -q firewalld
package firewalld is not installed
```

File: Templates/make_jenkins_master_standalone.tmplt.json (hunk 5 above)
Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: nvm, the point of the test is to confirm the file is there before installing it. 🤕

File: Templates/make_jenkins_master_standalone.tmplt.json (hunk 2 above)
Comment (ferricoxide) [inline Templates/make_jenkins_master_standalone.tmplt.json]: Alright, super-weird:

image

File: Templates/make_jenkins_master_standalone.tmplt.json (hunk 5 above)
Comment (ferricoxide) [inline Templates/make_jenkins_master_standalone.tmplt.json]: Yeah: was basically "make sure the cfn-staged file got staged; if so, install it to the target location".

The logic's there mostly with an eye towards changing to a method that doesn't require bouncing the `crond` service.

File: Templates/make_jenkins_master_standalone.tmplt.json (hunk 4 above)
Comment (ferricoxide) [inline Templates/make_jenkins_master_standalone.tmplt.json]: Eventually will be put into a count-1 ASG as a hedge against instance-loss.

File: Templates/make_jenkins_master_standalone.tmplt.json (hunk 2 above)
Comment (lorengordon) [inline Templates/make_jenkins_master_standalone.tmplt.json]: The problem is on the RHEL starting point AMIs. The ones provided by Amzn/Red Hat, not the ones we create.

**Pull request merged**